modules/command-engine/core/src/main/scala/surge/core/command/SurgeCommandModel.scala

```diff
@@ -21,6 +22,33 @@ private[surge] case class SurgeCommandKafkaConfig(
   override val eventsTopicOpt: Option[KafkaTopic] = if (publishStateOnly) None else Some(eventsTopic)
 }
 
+private[surge] object SurgeCommandModel {
+  def apply[AggId, Agg, Command, Event](
+      businessLogic: SurgeCommandBusinessLogicTrait[AggId, Agg, Command, Event]): SurgeCommandModel[Agg, Command, Nothing, Event] = {
+    new SurgeCommandModel[Agg, Command, Nothing, Event](
+      aggregateName = businessLogic.aggregateName,
+      kafka = businessLogic.kafkaConfig,
+      model = businessLogic.commandModel.toCore,
+      writeFormatting = businessLogic.writeFormatting,
+      readFormatting = businessLogic.readFormatting,
+      aggregateValidator = businessLogic.aggregateValidatorLambda,
+      metrics = businessLogic.metrics,
+      tracer = businessLogic.tracer)
+  }
+  def apply[AggId, Agg, Command, Rej, Event](
+      businessLogic: SurgeRejectableCommandBusinessLogicTrait[AggId, Agg, Command, Rej, Event]): SurgeCommandModel[Agg, Command, Rej, Event] = {
```

Comment:
Since it looks like these two `apply` methods are roughly the same and both `SurgeCommandBusinessLogicTrait` and `SurgeRejectableCommandBusinessLogicTrait` extend `SurgeGenericBusinessLogicTrait` it looks like we could use just one `apply` method that takes the base trait & can handle both of these implementations.

Reply:
We need both in order for [this](https://github.com/UltimateSoftware/surge/blob/220e1b137cc0b0c58d2c4cb3dd6c1196d399281c/modules/command-engine/javadsl/src/main/scala/surge/javadsl/command/SurgeCommand.scala#L44) line and the analogous line in the scaladsl to compile.